pkg/storage/tpr/storage_versioner_test.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tpr
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	_ "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/install"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/testapi"
+	metav1 "k8s.io/kubernetes/pkg/apis/meta/v1"
+	fakerestclient "k8s.io/kubernetes/pkg/client/restclient/fake"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/runtime/serializer"
+)
+
+func TestUpdateNonExistentObject(t *testing.T) {
+	v := getTestVersioner(
+		t,
+		func(*http.Request) (*http.Response, error) {
+			return getMockUpdateResponse(http.StatusNotFound), nil
+		},
+	)
+	// For convenience, we're passing 0 as the resource version, because we know
+	// that, internally, the function under test ONLY passes it to the core
+	// apiserver-- which is mocked out for the purposes of this test and will do
+	// nothing with it.
+	if err := v.UpdateObject(getTestBroker(), 0); err == nil {
+		t.Fatal("did not receive expected error")
+	}
+}
+
+func TestSuccessfulUpdateObject(t *testing.T) {
+	brokerObj := getTestBroker()
+	brokerCodec := getBrokerCodec(t)
+	versionedBrokerBytes, err := runtime.Encode(brokerCodec, brokerObj)
+	if err != nil {
+		t.Fatalf(
+			"error encoding broker object to versioned broker "+
+				"bytes: %s", err,
+		)
+	}
+	versionedBrokerObj, err := runtime.Decode(brokerCodec, versionedBrokerBytes)
+	if err != nil {
+		t.Fatalf(
+			"error decoding versioned broker bytes to versioned broker "+
+				"object: %s", err,
+		)
+	}
+	v := getTestVersioner(
+		t,
+		func(request *http.Request) (*http.Response, error) {
+			// Assert that what was sent to the mock apiserver was exactly the
```

Comment:
Ya. One refers to type version and another to instance (of the type) version. Agree... I don't know how to disambiguate that without getting really wordy.